meson.build

```diff
@@ -91,6 +92,20 @@ if modules_enabled
     cfg_var.set('ENABLE_MODULES', '1')
 endif
 
+# Detect and set symbol visibility
+if get_option('default_library') != 'static'
+    if host_os == 'windows' or host_os == 'cygwin'
+        cfg_var.set('DLL_EXPORT', true)
+        if cc.get_id() == 'msvc' or cc.get_id() == 'clang-cl'
+            cfg_var.set('_VIPS_PUBLIC', '__declspec(dllexport)')
+        elif cc.has_argument('-fvisibility=hidden')
+            cfg_var.set('_VIPS_PUBLIC', '__attribute__((visibility("default"))) __declspec(dllexport)')
+        endif
+    elif cc.has_argument('-fvisibility=hidden')
+        cfg_var.set('_VIPS_PUBLIC', '__attribute__((visibility("default")))')
+    endif
```

Comment:
I would set this regardless of default_library, as long as the host_os is some kind of non-Windows platform. It will never hurt, and may help the compiler optimize out some things even when building statically (I think).

Instead of cc.has_argument, you should use `cc.has_function_attribute('visibility:hidden')`.

Reply:
This was somewhat inspired by GLib:
https://github.com/GNOME/glib/blob/888ca6e36fddaf67b7a02e5ffabecec80f979d06/meson.build#L216-L234

I think(?) they excluded symbol visibility for static libraries because of the Windows-specific `DLL_EXPORT` definition (which controls whether the `DllMain` entry point should be present) and the `__declspec(dllexport)` thing (which is usually avoided for static libraries). But indeed, for non-Windows platforms, I think symbol visibility would also be useful for static libraries. (This made me wonder about [this hack](https://github.com/kleisauke/libvips-packaging/blob/meson/build/lin.sh#L471-L473), that is probably no longer necessary if symbol visibility is also set for static libraries in GLib)

Commit https://github.com/libvips/libvips/pull/2688/commits/dd5652f4033fb21c4d998c6042f1e82a24d961a4 resolves this.